### PR TITLE
Fix man proxy in FreeBSD

### PR DESCRIPTION
**Pull request by tesuji**
In FreeBSD, `man` command only [searches for man files in MANPATH][1]. 
Which means that FreeBSD `man` program does not accept interpret local man files.

On Bash-compatible shell, one could use this command to open rust specific manpages:

```sh
env MANPATH=$HOME/.rustup/toolchains/stable-x86_64-unknown-freebsd/share/man man rustc
```

[1]: https://www.freebsd.org/cgi/man.cgi?query=man&apropos=0&sektion=0&manpath=FreeBSD+12.0-RELEASE+and+Ports&arch=default&format=html

Locally tested on:
- [x] Ubuntu
- [x] FreeBSD 11

r? @kinnison 

**Note:** Ping me if someone needs to squash this PR.

### CODE REVIEW

File: src/cli/rustup_mode.rs [hunk 1]
```diff
@@ -1025,16 +1025,18 @@ fn doc(cfg: &Cfg, m: &ArgMatches<'_>) -> Result<()> {
...
     let toolchain = explicit_or_dir_toolchain(cfg, m)?;
     let mut man_path = toolchain.path().to_path_buf();
     man_path.push("share");
     man_path.push("man");
-    man_path.push("man1");
-    man_path.push(manpage.to_owned() + ".1");
-    utils::assert_is_file(&man_path)?;
+    utils::assert_is_directory(&man_path)?;
+    let mut man_path = man_path.into_os_string();
+    man_path.push(":");
```
Comment (lzutao) [inline src/cli/rustup_mode.rs]: MANPATH is separated by `:`. Man command needs it as it is.

File: src/cli/rustup_mode.rs [hunk 2]
```diff
@@ -1025,16 +1025,17 @@ fn doc(cfg: &Cfg, m: &ArgMatches<'_>) -> Result<()> {
...
-    let mut man_path = toolchain.path().to_path_buf();
-    man_path.push("share");
-    man_path.push("man");
-    man_path.push("man1");
-    man_path.push(manpage.to_owned() + ".1");
-    utils::assert_is_file(&man_path)?;
+    let mut manpaths = toolchain.path().to_path_buf();
+    manpaths.push("share");
+    manpaths.push("man");
+    utils::assert_is_directory(&manpaths)?;
```
Comment (tesuji) [inline src/cli/rustup_mode.rs]: Is it necessary to check for existence of the man dir?
Can we assume that Rustup always install manpages in this location?

**Comment by lzutao** [comment]
I think Daniel is tired. So I pick @dwijnand as reviewer.

File: src/cli/rustup_mode.rs (hunk 2 above)
Comment (kinnison) [reply src/cli/rustup_mode.rs]: I think it's not unreasonable to fail more informatively if someone types `rustup man` and the manpages moved, the error might be more helpful than a simply missing or incorrect manpage

**Review by kinnison — COMMENTED** [review]
I think this looks good, though I'm loathe to approve because `-M` overrides the `MANPATH` environment variable which might also be set and relevant.  I'd be more comfortable with manipulating that than passing `-M`

**Comment by tesuji** [comment]
@kinnison So you want to manipulate `MANPATH` env var via [`Command::env`](https://doc.rust-lang.org/std/process/struct.Command.html#method.env)?

**Comment by kinnison** [comment]
I figure that if `MANPATH` is set, we have to do that to be correct (i.e. prepend our path onto `MANPATH`)   If it's not set, we could use `-M` but at that point the code is simpler to always prepend to `MANPATH`

**Comment by tesuji** [comment]
@kinnison I pushed the change you requested.

**Review by kinnison — COMMENTED** [review]
This looks reasonable to me.  Have you verified it against FreeBSD and at least one Linux?

**Comment by kinnison** [comment]
If so, please rebase to clean up the history of the branch before I merge.

**Comment by tesuji** [comment]
@kinnison I tested on Ubuntu. FreeBSD I only test the equivalent shell command.

**Review by kinnison — APPROVED** [review]
LGTM

**Pull request merged**